### DISCUSSION TITLE
How re-calculate the count of detail when using RelationManager ?

### DISCUSSION THREAD

**Discussion by altimusin** [filamentphp/filament#5017] [Help] [answered]
Hello,

I made CommentsRelationManager for PostResource.

How I can refresh/recalculate the count value on header, whatever the comments is added or deleted.

![image](https://user-images.githubusercontent.com/43429317/202954999-96a1d07d-7a2e-40a3-8736-4e982ac26a28.png)
Image note : I add a new comment, but the count value not refreshing.

This is my PostResource Form:

```
    public static function form(Form $form): Form
    {
        return $form
            ->schema([
                TextInput::make('title')->required(),
                TextInput::make('body')->required(),
                TextInput::make('count')
                    ->reactive()
                    ->label('count')
                    ->disabled()
                    ->placeholder(
                        function ($state, $record) {
                            return Comment::where('post_id', $record->id)->get()->count();
                        }
                    )
            ]);
    }
```

**Comment by danharrin** [comment, marked as answer]
Emit a Livewire event from the relation manager `after()` the Edit or Create action, and register a listener on the page, calling `$refresh`. This will refresh the value of the placeholder.

**Reply by danharrin** [reply]
Also, you can just use `->placeholder(fn ($record) => $record->comments()->count())`

**Reply by altimusin** [reply]
Hi danharrin,

Really appreciate your help. But I don't understand how to calling $refresh on Edit or Create action.
Can you elaborate more ? please..

Thank you
```
class CommentsRelationManager extends RelationManager
{ ...
    public static function table(Table $table): Table
    {
        return $table
            ->columns([
                Tables\Columns\TextColumn::make('comment'),
            ])
            ->filters([
                //
            ])
            ->headerActions([
                Tables\Actions\CreateAction::make(),
            ])
            ->actions([
                Tables\Actions\EditAction::make()->after( ???? ),
                Tables\Actions\DeleteAction::make(),
            ])
            ->bulkActions([
                Tables\Actions\DeleteBulkAction::make(),
            ]);
    }
}
```

**Reply by danharrin** [reply]
```php
Tables\Actions\EditAction::make()->after(fn ($livewire) => $livewire->emit('refreshForm'))
```

In EditPost.php

```php
protected $listeners = [
    'refreshForm' => '$refresh',
];
```

**Reply by altimusin** [reply]
Hi danharrin,

It give me error result : Unable to resolve dependency [Parameter #2 [ <required> $refresh ]] in class App\Filament\Resources\PostResource
![image](https://user-images.githubusercontent.com/43429317/203433618-c4b0c0f4-a174-4b58-9aeb-1e00cd64b810.png)

```
class EditPost extends EditRecord
{
    protected static string $resource = PostResource::class;

    protected $listeners = [
        'refreshForm' => '$refresh',
    ];

    protected function getActions(): array
    {
        return [
            Actions\ViewAction::make(),
            Actions\DeleteAction::make(),
        ];
    }
```

**Reply by danharrin** [reply]
And what code for your relation manager?

**Reply by altimusin** [reply]
Hi, @danharrin 

My fault, I put wrong code,.. Now It solved.

Thank you.
